lib/writer.js

```diff
@@ -1,64 +1,29 @@
-const common = require('asset-pipe-common');
-const uglify = require('uglify-js');
+const outputFeed = require('./browserify-plugins/output-feed');
+const idHasher = require('./browserify-plugins/id-hasher');
+const assert = require('assert');
+const { name } = require('../package.json');
 
 module.exports = class Writer {
-    constructor(files = [], options = {}, bundle = false, minify = false) {
-        const browserify = new Browserify(files, options);
-        const dictionary = {};
-
-        if (bundle) {
-            return browserify;
-        }
-
-        const hasher = new stream.Transform({
-            objectMode: true,
-            transform(obj, enc, next) {
-                dictionary[obj.id] = common.hasher(obj.source);
-                this.push(obj);
-                next();
-            },
-        });
-        browserify.pipeline.get('deps').push(hasher);
-
-        const labeler = new stream.Transform({
-            objectMode: true,
-            transform(obj, enc, next) {
-                obj.id = dictionary[obj.id];
-                Object.keys(obj.deps).forEach(key => {
-                    obj.deps[key] = dictionary[obj.deps[key]];
-                });
-                this.push(obj);
-                next();
-            },
-        });
-        browserify.pipeline.get('label').splice(0, 1, labeler);
-
-        // Replace the step in browserifys build pipeline which convert the
-        // internal dependency object structure to a browser bundle transform
-        // which just pipes objects through. This leave us with a stream of
-        // dependency objects we can pipe anywhere
-
-        const packer = new stream.Transform({
-            objectMode: true,
-            transform(obj, enc, next) {
-                if (minify) {
-                    const minified = uglify.minify(obj.source, {
-                        fromString: true,
-                    });
-                    if (minified.code) {
-                        obj.source = minified.code;
-                    }
-                }
-                this.push(obj);
-                next();
-            },
-        });
-        browserify.pipeline.get('pack').splice(0, 1, packer);
-
-        return browserify;
+    constructor(files, options = {}, bundle = false) {
+        assert(
+            Array.isArray(files) || typeof files === 'string',
+            `Expected 'files' argument to ${name} constructor to be either a string or an array. Instead got '${typeof files}'`
+        );
+        assert(
+            typeof options === 'object',
+            `Expected optional 'options' argument to ${name} constructor to be an object. Instead got '${typeof options}'`
+        );
+        assert(
+            typeof bundle === 'boolean',
+            `Expected optional 'bundle' argument to ${name} constructor to be a boolean. Instead got '${typeof bundle}'`
+        );
+
+        const opts = { dedupe: false, ...options };
```

Comment:
should we allow users to overwrite this?

Reply:
"this" being `dedupe`

Reply:
probably not eh?